depthai_ros_driver/CMakeLists.txt

```diff
@@ -1,171 +1,81 @@
 
-set (PKG_DEPS_COMMON
-    camera_info_manager
-    cv_bridge
-    depthai_ros_msgs
-    image_transport
-    sensor_msgs
-    std_msgs
+set(PKG_DEPS
+  camera_info_manager
+  cv_bridge
+  depthai_ros_msgs
+  image_transport
+  sensor_msgs
+  std_msgs
 )
 
-#==============================================================================
-# Compile in ROS1
-#==============================================================================
-
 if($ENV{ROS_VERSION} STREQUAL 1)
-
-set (PKG_DEPS
-    ${PKG_DEPS_COMMON}
+  set (PKG_DEPS
+    ${PKG_DEPS}
     nodelet
     roscpp
     node_interface
-)
-
-find_package (catkin REQUIRED COMPONENTS ${PKG_DEPS})
-
-# Add depthai-core dependency
-add_subdirectory (${DEPTHAI_CORE} depthai-core)
-
-catkin_package (
-    INCLUDE_DIRS include
-    LIBRARIES ${PROJECT_NAME}_nodelet
-    CATKIN_DEPENDS ${PKG_DEPS}
-    DEPENDS Boost
-)
-
-###########
-## Build ##
-###########
-
-include_directories (include)
-
-include_directories (SYSTEM
-    ${Boost_INCLUDE_DIRS}
-    ${OpenCV_INCLUDE_DIRS}
-    ${catkin_INCLUDE_DIRS}
-)
-
-# internal library, not to be exported
-add_library (${PROJECT_NAME}_lib src/depthai_base.cpp src/depthai_common.cpp)
-target_link_libraries (${PROJECT_NAME}_lib
-    ${catkin_LIBRARIES}
-    ${OpenCV_LIBRARIES}
-    ${Boost_LIBRARIES}
-    depthai-core
-)
-add_dependencies (${PROJECT_NAME}_lib
-    ${${PROJECT_NAME}_EXPORTED_TARGETS} ${catkin_EXPORTED_TARGETS}
-)
-
-add_library (${PROJECT_NAME}_nodelet src/depthai_nodelet.cpp)
-target_link_libraries (${PROJECT_NAME}_nodelet
-    ${PROJECT_NAME}_lib
-)
-add_executable (${PROJECT_NAME} src/depthai_node.cpp)
-target_link_libraries (${PROJECT_NAME}
-    ${PROJECT_NAME}_lib
-)
-
-#############
-## Install ##
-#############
-
-install (TARGETS ${PROJECT_NAME}_lib ${PROJECT_NAME}_nodelet ${PROJECT_NAME}
-  ARCHIVE DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
-  LIBRARY DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
-  RUNTIME DESTINATION ${CATKIN_PACKAGE_BIN_DESTINATION}
-)
-
-# Not installing the headers because this is not a library package
-install (DIRECTORY launch resources scripts urdf params
-    DESTINATION ${CATKIN_PACKAGE_SHARE_DESTINATION}
-    USE_SOURCE_PERMISSIONS
-)
-
-install (FILES nodelet_plugins.xml DESTINATION ${CATKIN_PACKAGE_SHARE_DESTINATION})
-RETURN()
+  )
 else()
-  message(VERBOSE "ROS version is $ENV{ROS_VERSION}, skipping ROS1 ${PROJECT_NAME}")
+  set (PKG_DEPS
+    ${PKG_DEPS}
+    rclcpp
+    rclcpp_components
+  )
 endif()
 
+include(cmake/ros_version_agnostic_add_node.cmake)
+
+# A bug in the CMake of the included external project depthai-core clobbers the
+# paths used to find ROS packages, so we have to search for them first and then
+# add the external project
+find_dependent_packages(${PKG_DEPS})
+
+add_subdirectory(${DEPTHAI_CORE} depthai-core)
+
+add_ros_node(depthai_ros_driver
+  SKIP_PACKAGE_SEARCH
+  PROJECT_NAME ${PROJECT_NAME}
+  COMPONENT_CLASS "rr::DepthAIBaseRos2"
```

Comment:
Is it better if we name it as `ROS2_COMPONENT_CLASS` and place it below `ROS1_COMPONENT_SOURCES` for readability

Reply:
I've renamed the parameter. The ordering doesn't matter as it's a keyword parameter.